src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SubjectIdentifier.cs

```diff
@@ -97,6 +97,40 @@ internal SubjectIdentifier(SignerIdentifierAsn signerIdentifierAsn)
 
         public SubjectIdentifierType Type { get; }
         public object Value { get; }
+
+        public bool MatchesCertificate(X509Certificate2 certificate)
+        {
+            switch (Type)
+            {
+                case SubjectIdentifierType.IssuerAndSerialNumber:
+                    {
+                        X509IssuerSerial issuerSerial = (X509IssuerSerial)Value;
+                        byte[] serialNumber = issuerSerial.SerialNumber.ToSerialBytes();
+                        string issuer = issuerSerial.IssuerName;
+                        byte[] certSerialNumber = certificate.GetSerialNumber();
+                        if (Helpers.AreByteArraysEqual(certSerialNumber, serialNumber) && certificate.Issuer == issuer)
+                            return true;
```

Comment:
Why is this an if instead of a direct return?

If it makes sense to be an if, add braces, since it's not in the direct method scope.

(And it would make me happy to have a blank line before the `if` statement, because I'm compulsive like that).

Reply:
It was in a loop in the previous version